Source_Files/RenderOther/HUDRenderer_Lua.cpp

```diff
@@ -53,9 +54,32 @@ HUD_Lua_Class *Lua_HUDInstance()
 void Lua_DrawHUD(short time_elapsed)
 {
 	HUD_Lua.update_motion_sensor(time_elapsed);
+    
+    #ifdef HAVE_OPENGL
+    //bool isEnabled_GT2 = glIsEnabled (GL_TEXTURE_2D); //NOT SUPPORTED ANGLE ENUM
+    bool isEnabled_GCF = glIsEnabled (GL_CULL_FACE);
+    bool isEnabled_GDT = glIsEnabled (GL_DEPTH_TEST);
+    //bool isEnabled_GAT = glIsEnabled (GL_ALPHA_TEST); //NOT SUPPORTED ANGLE ENUM
+    bool isEnabled_GST = glIsEnabled (GL_STENCIL_TEST);
+    bool isEnabled_GB = glIsEnabled (GL_BLEND);
+    //bool isEnabled_GF = glIsEnabled (GL_FOG); //NOT SUPPORTED ANGLE ENUM
+    int previousMode = MSI()->currentActiveMode();
+    #endif
+    
 	HUD_Lua.start_draw();
 	L_Call_HUDDraw();
 	HUD_Lua.end_draw();
+    
+    #ifdef HAVE_OPENGL
+    MSI()->matrixMode(previousMode);
+    //if ( isEnabled_GT2 ) { glEnable ( GL_TEXTURE_2D ) ; } else { glDisable ( GL_TEXTURE_2D ); } //NOT SUPPORTED ANGLE ENUM
+    if ( isEnabled_GCF ) { glEnable ( GL_CULL_FACE ) ; } else { glDisable ( GL_CULL_FACE ); }
+    if ( isEnabled_GDT ) { glEnable ( GL_DEPTH_TEST ) ; } else { glDisable ( GL_DEPTH_TEST ); }
+    //if ( isEnabled_GAT ) { glEnable ( GL_ALPHA_TEST ) ; } else { glDisable ( GL_ALPHA_TEST ); } //NOT SUPPORTED ANGLE ENUM
+    if ( isEnabled_GST ) { glEnable ( GL_STENCIL_TEST ) ; } else { glDisable ( GL_STENCIL_TEST ); }
+    if ( isEnabled_GB )  { glEnable ( GL_BLEND ) ; } else { glDisable ( GL_BLEND ); }
+    //if ( isEnabled_GF )  { glEnable ( GL_FOG ) ; } else { glDisable ( GL_FOG ); } //NOT SUPPORTED ANGLE ENUM
+    #endif
```

Comment:
Those OpenGL functions will make the game crash when running in software mode when compiled with OpenGL support (always the cast since master isn't compilable without OpenGL support for now)